src/node_data.cpp

```diff
@@ -209,12 +206,9 @@ node* node_data::get(node& key, shared_memory_holder /* pMemory */) const {
     return nullptr;
   }
 
-  for (node_map::const_iterator it = m_map.begin(); it != m_map.end(); ++it) {
-    if (it->first->is(key))
-      return it->second;
-  }
-
-  return nullptr;
+  auto it = std::find_if(m_map.begin(), m_map.end(), [&](const std::pair<node*, node*> node)
```

Comment:
I think this still reads better as a for loop, but perhaps you can change it to:

for (const auto& kv : m_map) {
  if (kv.first->is(key)) {
    return it->second;
  }
}
return nullptr;

Reply:
Sure. Using algorithms seems to be preferred by some. They result in better code generation sometimes. I’ve seen that with std::accumulate.